distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso

```diff
@@ -0,0 +1,586 @@
+
+    ## GROUP Operators
+       ICON math
+       Divides a `Decimal` by another `Decimal` or other kind of number, or
+       divides another kind of number by a `Decimal`.
+
+       Arguments:
+       - that: The number to divide by this.
+
+       Division in Enso will undergo automatic conversions such that you need
+       not convert other numeric types to `Decimal` manually.
+
+       > Example
+         Dividing 1065.9378 by 23.34.
+
+             a = Decimal.new "1065.9378"
+             b = Decimal.new "23.34"
+             c = a / b
+             # => Decimal.new 45.67
+    / : Decimal -> Decimal -> Decimal
+    / self (that:Decimal) = self.divide that
+
+    ## GROUP Operators
+       Compute the negation of this.
+
+       > Example
+         Negate 5.1 to get -5.1.
+
+             5.1.negate
+             # => Decimal.new -5.1
+    negate : Decimal
+    negate self = Decimal.Value self.big_decimal.negate
+
+    # PRIVATE
+    precision : Integer
+    precision self = self.big_decimal.precision
+
+    # PRIVATE
+    scale : Integer
+    scale self = self.big_decimal.scale
+
+    # PRIVATE
+    unscaled_value : Integer
+    # TODO: This +0 is needed to allow internal represeantations to be compared.
+    unscaled_value self = self.big_decimal.unscaledValue + 0
+
+    # PRIVATE
+    internal_representation : [Integer]
+    internal_representation self = [self.unscaled_value, self.precision, self.scale]
+
+    ## PRIVATE
+    to_text : Text
+    to_text self = self.big_decimal.toString
+
+    ## PRIVATE
+    to_text_without_scientific_notation : Text
+    to_text_without_scientific_notation  self = self.big_decimal.toPlainString
+
+## PRIVATE
+handle_number_format_exception ~action =
+    Panic.catch NumberFormatException action caught_panic->
+        Error.throw (Number_Parse_Error.Error caught_panic.payload.getMessage)
+
+## PRIVATE
+handle_precision_loss : Any -> Conversion_Result -> Any
+handle_precision_loss original_value conversion_result:Conversion_Result -> Any =
+    if conversion_result.hasPrecisionLoss.not then conversion_result.newValue else
+        new_value = conversion_result.newValue
+        Warning.attach (Loss_Of_Numeric_Precision.Warning original_value new_value) new_value
+
+## PRIVATE
+attach_loss_of_numeric_precision : Float -> Any -> Any
+attach_loss_of_numeric_precision f value =
+    Warning.attach (Loss_Of_Numeric_Precision.Warning f f) value
+
+handle_java_exception ~action =
+    Panic.catch ArithmeticException action caught_panic->
+        Error.throw (Arithmetic_Error.Error caught_panic.payload.getMessage)
+
+## PRIVATE
+type Decimal_Comparator
+    #compare x:Decimal y:Decimal =
+    compare x:Decimal y:Decimal =
+        Ordering.from_sign (x.big_decimal.compareTo y.big_decimal)
+    hash x:Decimal = Decimal_Utils.hashCodeOf x.big_decimal
+
+## PRIVATE
+Comparable.from (_:Decimal) = Decimal_Comparator
+Comparable.from (_:Number) = Decimal_Comparator
```

Comment:
The first conversion is marked as PRIVATE, but the second one is not. Let's be consistent.

Reply:
Done